ets: improve ets:new/2 already_exists named_table error

gonzalobf opened the pull request:
Adds extra information in `erl_db.c` before returning the error for already used table name. The error is only returned if the option `named_table` is give and the name is already in use.

The extra information is used later to format the error message.

Closes #6252

gonzalobf commented:
I tried only changing the function in `erl_stdlib_errors`  by adding something like `ets:whereis(Name)` so I could check if the name was already in use, but the option only works if the `ets` is created in another process. If both tables are created under the same process, when the `format_ets_error` function is called the first table has been already destroyed and doesn't exist anymore.

Since I'm touching `erl_db.c` and adding an atom, I suspect it shouldn't be based on `maint` but let me know if I'm wrong and I'll change it.

sverker commented:
Thanks for the contribution.

I have pushed three additional commits here:
https://github.com/sverker/otp/tree/gonzalobf/gonzalo/ets/already-exists/GH-6252

I propose you
1. Add them to your branch for this PR (I did not have access rights to do so).
2. Remove your `ets_SUITE:already_exists`, I think it was made redundant by my extension of `ets_SUITE:error_info`.
3. If you are satisfied, squash all of it into one commit.
4. Rebase the resulting branch on maint: `git rebase --onto maint master`
5. Edit the PR to be based on maint.

Bug fixes are normally done on maint, especially if they are low risk and without backward compatibility problems.

gonzalobf commented:
@sverker Thank you for your help. Your changes make sense so I added to my branch and squash them. I also deleted the ets_SUITE test and changed the PR be based in `maint` as you suggested.

I'm happy to do more modifications if there is something missing or anything that can be improved

sverker commented:
@bjorng I will merge this unless you see any problems.

sverker commented:
Merged to maint for OTP 25.2

Thanks for the contribution.
